File: Java/Keyboard.java
public class Keyboard extends Peripheral {
    private String layoutType;
    private boolean mechanical;
    
    public Keyboard() {
        super();
        this.layoutType = "";
        this.mechanical = false;
    }
    
    public Keyboard(String layoutType, boolean mechanical, String merk, String nama, double harga,
                   String tipePeripheral, String koneksi) {
        super(merk, nama, harga, tipePeripheral, koneksi);
        this.layoutType = layoutType;
        this.mechanical = mechanical;
    }
    
    public Keyboard(String layoutType, boolean mechanical, String merk, String nama, double harga) {
        this(layoutType, mechanical, merk, nama, harga, "Keyboard", "USB");
    }
    
    public String getLayoutType() {
        return layoutType;
    }
    
    public void setLayoutType(String layoutType) {
        this.layoutType = layoutType;
    }
    
    public boolean isMechanical() {
        return mechanical;
    }
    
    public void setMechanical(boolean mechanical) {
        this.mechanical = mechanical;
    }
    
    @Override
    public void tampilkanInfo() {
        System.out.println(tipePeripheral + ": " + merk + " " + nama + 
                           " (" + layoutType + ", " + (mechanical ? "Mechanical" : "Membrane") + ")");
        System.out.println("    Koneksi: " + koneksi + ", Harga: Rp" + harga);
    }
}

File: Python/Keyboard.py
# keyboard.py
from Peripheral import Peripheral

class Keyboard(Peripheral):
    """Keyboard peripheral class"""
    
    def __init__(self, layout_type="", mechanical=False, merk="", nama="", harga=0,
                 tipe_peripheral="Keyboard", koneksi="USB"):
        super().__init__(merk, nama, harga, tipe_peripheral, koneksi)
        self.layout_type = layout_type
        self.mechanical = mechanical
    
    def get_layout_type(self):
        return self.layout_type
    
    def set_layout_type(self, layout_type):
        self.layout_type = layout_type
    
    def is_mechanical(self):
        return self.mechanical
    
    def set_mechanical(self, mechanical):
        self.mechanical = mechanical
    
    def tampilkan_info(self):
        print(f"{self.tipe_peripheral}: {self.merk} {self.nama} ({self.layout_type}, {'Mechanical' if self.mechanical else 'Membrane'})")
        print(f"    Koneksi: {self.koneksi}, Harga: Rp{self.harga}")